xbmc/pvr/channels/PVRChannelProvider.h

```diff
@@ -0,0 +1,338 @@
+
+    /*!
+      * @brief Get this provider's language
+      * @return This provider's language
+      */
+    std::string GetLanguage() const;
+
+    /*!
+     * @brief Set the language
+     * @param strLanguage The new language for this provider
+     * @param bIsUserSetLanguage True is the user is setting this language via GUI, false otherwise
+     * @return true if the language was updated successfully
+     */
+    bool SetLanguage(const std::string& strLanguage, bool bIsUserSetLanguage = false);
+
+    /*!
+      * @brief Get if this provider's has thumb image path. Note only PVR add-on providers.
+      * @return True is this add-on provider has a thumb image path, false otherwise.
+      */
+    bool HasThumbPath() const;
+
+    /*!
+      * @brief Get this provider's thumb image path. Note only PVR add-on providers will set this value.
+      * @return This add-on provider's thumb image path.
+      */
+    std::string GetThumbPath() const { return m_strThumbPath; }
+
+    /*!
+      * @brief Whether a provider is a default provider of a PVR Client add-on or not
+      * @return True is this provider is of a PVR Client add-on, false otherwise.
+      */
+    bool IsClientProvider() const { return m_bIsClientProvider; }
+
+    /*!
+     * @brief The path in the XBMC VFS to be used by PVRManager for this provider
+     * @return The path in the XBMC VFS to be used by PVRManager for this provider
+     */
+    std::string GetPath() const { return m_strPath; }
+
+    /*!
+     * @brief Recreate the path for this provider based on current provider name
+     */
+    void UpdatePath();
+
+    /*!
+     * @brief updates this provider from the provided entry
+     * @param fromProvider A provider containing the data that shall be merged into this provider's data.
+     * @param updateMode update as User, Client or DB
+     * @return true if the provider was updated successfully
+     */
+    bool UpdateEntry(const std::shared_ptr<CPVRChannelProvider>& fromProvider,
+                     ProviderUpdateMode updateMode);
+
+    /*!
+     * @brief Does this provider have any user set values
+     * @return true if the provider contains user set values, false otherwise
+     */
+    bool HasUserSetValues();
+
+    /*!
+     * @brief Persist this provider in the local database.
+     * @param updateRecord True if an existing record should be updated, false for an insert
+     * @return True on success, false otherwise.
+     */
+    bool Persist(bool updateRecord = false);
+
+    /*!
+     * @brief Delete this provider from the local database.
+     * @return True on success, false otherwise.
+     */
+    bool DeleteFromDatabase();
+
+    mutable CCriticalSection m_critSection;
+
+  private:
+    CPVRChannelProvider(const CPVRChannelProvider& provider) = delete;
+    CPVRChannelProvider& operator=(const CPVRChannelProvider& orig) = delete;
+
+    int m_iProviderId; /*!< the identifier given to this channel provider by the TV database */
```

Comment:
You initialize some members here, some not. Is this intended?

Reply:
Moved initialisation to the constructors.

Reply:
done